python/Ganga/Core/GangaRepository/GangaRepositoryXML.py

```diff
@@ -291,12 +291,12 @@ def index_load(self, this_id):
                     obj = self._make_empty_object_(this_id, cat, cls)
                 except Exception as err:
                     raise IOError('Failed to Parse information in Index file: %s. Err: %s' % (fn, err))
-            this_cache = obj._index_cache
+            this_cache = obj._index_cache_dict
```

Comment:
The getter method completely subverts the point of this code by checking that we get the same thing unless the object is not loaded. We know her the object has _not_ been loaded and in other tests the object _has just been loaded_. We know this because this class is in charge of doing this. Using the getter gets you into a strange state depending on whether the Registry/Repo is saying the object is in memory _right now_ which can potentially get out of sync during object load.
(In addition to calling potentially lockable functions higher up which don't need to be done.)